sympy/logic/boolalg.py

```diff
@@ -527,9 +525,9 @@ def diff(self, *symbols, **assumptions):
         return Derivative(self, *symbols, **assumptions)
 
     def _eval_derivative(self, x):
-        from sympy.core.relational import Eq
-        from sympy.functions.elementary.piecewise import Piecewise
         if x in self.binary_symbols:
+            from sympy.core.relational import Eq
```

Comment:
No need to import unless this branch is used.

Reply:
It would be good to resolve the circular dependency between `sympy.logic.boolalg` and `sympy.core.relational`. Maybe `boolalg.py` really belongs in the core or maybe `Relational` belongs in the logic module.